src/OpenTelemetry.Instrumentation.Dependencies/Implementation/HttpHandlerDiagnosticListener.cs

```diff
@@ -75,8 +75,7 @@ public override void OnStartActivity(Activity activity, object payload)
                 return;
             }
 
-            // TODO: Avoid the reflection hack once .NET ships new Activity with Kind settable.
-            activity.GetType().GetProperty("Kind").SetValue(activity, ActivityKind.Client);
+            activity.SetKind(ActivityKind.Client);
```

Comment:
Can you check GrpcClientDiagnosticListener as it also has to set Kind.

Reply:
done! thank you!